library/src/main/java/com/novoda/downloadmanager/lib/DownloadService.java

```diff
@@ -259,7 +250,7 @@ public boolean handleMessage(@NonNull Message msg) {
             // TODO: handle media scanner timeouts
 
             final boolean isActive;
-            synchronized (downloads) {
+            synchronized (DownloadService.this) {
```

Comment:
wasn't sure, the entire call back is done on a single handler thread, shouldn't need to be synchronised in the first place